### PR TITLE
Adds Bitcraze Crazyflie2

### PR DESCRIPTION
**Pull request by varadVaidya**
This PR adds a crayflie description. I have attempted to follow contribution guidelines. In README.md you can find notes on the process of converting the ROS description to MJCF

What help I need:
License: As mentioned in the README, i have picked the URDF and mesh file from [crazyflie_ros](https://github.com/whoenig/crazyflie_ros), which has MIT License. So does this work be under MIT too?

### CODE REVIEW

File: bitcraze_crazyflie_2/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Bitcraze Crazyflie2 Description (MJCF)
+
+Tested with MuJoCo 3.1.2
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:3]: Can you change this to the minimum version required?

File: bitcraze_crazyflie_2/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Overview
+
+This package contains  simplified robot description (MJCF) of the Crazyflie2 model from the [Bitcraze](https://www.bitcraze.io/). It is derived from the publicly available, but archived, ROS description: [crazyflie_ros](https://github.com/whoenig/crazyflie_ros).
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:7]: Crazyflie2 model from Bitcraze (remove the). Also no need to mention the repo is archived.

File: bitcraze_crazyflie_2/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+![Crazyflie2 Model](cf2.png)
+
+## URDF → MJCF Conversion
+
+1. Convereted the DAE mesh file in `crazyflie_description` to OBJ format using [Blender](https://www.blender.org/).
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:12]: Typo fix: converted

File: bitcraze_crazyflie_2/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+2. Processed the OBJ file with [obj2mjcf](https://github.com/kevinzakka/obj2mjcf)
+3. Added a `<freejoint>` to the root body, and some lighting in the XML file.
+4. Set the inertial properties to values obtained from the datasheet and [MIT's system identification](https://groups.csail.mit.edu/robotics-center/public_papers/Landry15.pdf)
+5. These properties are set via inertial tag i.e. `pos ="0 0 0" mass="0.027" diaginertia="2.3951e-5 2.3951e-5 3.2347e-5"`
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:16]: Make bullet point number 5 a sub-bullet point of 4.

File: bitcraze_crazyflie_2/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+6. Added combined thrust and body moments about a `site` placed at the inertial frame. The `ctrlrange` limits are currently arbitrary and need to be further tuned.
+7. Added `scene.xml` which includes the quadrotor, with a textured ground plane and skybox.
+
+## License
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:20]: The source repo is MIT so this license should be MIT (make sure you copy the MIT license as a file to in this directory).

File: bitcraze_crazyflie_2/cf2.xml [hunk 6]
```diff
@@ -0,0 +1,119 @@
+<mujoco model="cf2">
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:1]: Add newlines between the different sections of the XML (e.g., between option and compiler and default etc.).

File: bitcraze_crazyflie_2/cf2.xml [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  <option integrator="RK4"/>
+  <compiler inertiafromgeom="false" coordinate="local" meshdir="cf2"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:3]: coordinate="local" is the default (in fact global is now deprecated) so please remove.

File: bitcraze_crazyflie_2/cf2.xml [hunk 8]
```diff
@@ -0,0 +1,119 @@
+<mujoco model="cf2">
+  <option integrator="RK4"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:2]: Is RK4 necessary?

File: bitcraze_crazyflie_2/cf2.xml [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+  <compiler inertiafromgeom="false" coordinate="local" meshdir="cf2"/>
+  <default>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:4]: Can you add fluid and density parameters [as in the X2](https://github.com/google-deepmind/mujoco_menagerie/blob/main/skydio_x2/x2.xml#L4)?

File: bitcraze_crazyflie_2/cf2.xml [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    <default class="visual">
+      <geom group="2" type="mesh" contype="0" conaffinity="0"/>
+    </default>
+    <default class="collision">
+      <geom group="3" type="mesh"/>
+    </default>
+  </default>
+  <asset>
+    <material name="Material10-polished_plastic1_001" specular="0.5" shininess="0.225"
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:13]: Can you give the material more readable names?

* `Material10-polished_plastic1_001` -> `polished_plastic`

Also, can you format the material rgbas to no more than 2-3 decimal places?

File: bitcraze_crazyflie_2/cf2.xml [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+      rgba="0.968627 0.878431 0.600000 1.000000"/>
+    <material name="Material14-green_medium_gloss_plastic_001" specular="0.5" shininess="0.225"
+      rgba="0.109804 0.184314 0.000000 1.000000"/>
+    <material name="Material15_001" specular="0.5" shininess="0.225" rgba="0.792157 0.819608 0.933333 1.000000"/>
+    <material name="Material3-polypropylene2_001" specular="0.5" shininess="0.225"
+      rgba="1.000000 1.000000 1.000000 1.000000"/>
+    <material name="Material6-black_medium_gloss_plastic_001" specular="0.5" shininess="0.225"
+      rgba="0.101961 0.101961 0.101961 1.000000"/>
+    <material name="Material8-burnished_chrome_001" specular="0.5" shininess="0.225"
+      rgba="0.898039 0.898039 0.898039 1.000000"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:25]: Add a newline between material and mesh sections please.

File: bitcraze_crazyflie_2/cf2.xml [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+    <mesh file="cf2_collision_27.obj"/>
+    <mesh file="cf2_collision_28.obj"/>
+    <mesh file="cf2_collision_29.obj"/>
+    <mesh file="cf2_collision_30.obj"/>
+    <mesh file="cf2_collision_31.obj"/>
+  </asset>
+  <worldbody>
+    <body name="cf2" pos="0 0 0.1">
+      <inertial pos="0 0 0" mass="0.027" diaginertia="2.3951e-5 2.3951e-5 3.2347e-5"/>
+      <joint name="root" type="free" damping="0" armature="0" pos="0 0 0" limited="false"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:69]: You can use the `freejoint` tag as a shortcut here.

File: bitcraze_crazyflie_2/cf2.xml [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+      <geom mesh="cf2_collision_28" class="collision"/>
+      <geom mesh="cf2_collision_29" class="collision"/>
+      <geom mesh="cf2_collision_30" class="collision"/>
+      <geom mesh="cf2_collision_31" class="collision"/>
+
+      <site name="actuation" type="sphere" size="0.001 0.001 0.001" rgba="1 0 0 1" pos="0 0 0"/>
+    </body>
+  </worldbody>
+  <actuator>
+    <motor ctrllimited="true" ctrlrange="0 0.35" gear="0 0 1 0 0 0" site="actuation" name="body_thrust"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:114]: Add `autolimits=true` in the compiler and then remove `ctrllimited` from all the actuators.

**Review by kevinzakka — CHANGES_REQUESTED** [review]
This is a great start, thanks! Left some comments to address, plus the below summary:

* Overall, I think it would be helpful if you looked at the [Skydio X2 XML](https://github.com/google-deepmind/mujoco_menagerie/blob/main/skydio_x2/x2.xml) and then tried to copy some of the settings there (e.g. density and viscosity, integrator, etc.)
* Please remove the MTL files, they are not used by MuJoCo.
* Remove the stray `.pyc` file.

File: bitcraze_crazyflie_2/cf2.xml (hunk 8 above)
Comment (varadVaidya) [reply bitcraze_crazyflie_2/cf2.xml:2]: Using RK4 is not necessary here, but i feel its a better choice. Considering the small inertial properties, if any down wash models or external disturbances are implemented, then a stable integrator might provide a more accurate simulation.

File: bitcraze_crazyflie_2/LICENSE [hunk 14]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2024 varadVaidya
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/LICENSE:3]: @varadVaidya This should preserve the original copyright name from the ROS repo.

File: bitcraze_crazyflie_2/README.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+2. Processed the OBJ file with [obj2mjcf](https://github.com/kevinzakka/obj2mjcf)
+3. Added a `<freejoint>` to the root body, and some lighting in the XML file.
+4. Set the inertial properties to values obtained from the datasheet and [MIT's system identification](https://groups.csail.mit.edu/robotics-center/public_papers/Landry15.pdf)
+    * These properties are set via inertial tag i.e. `pos ="0 0 0" mass="0.027" diaginertia="2.3951e-5 2.3951e-5 3.2347e-5"`
+5. Added combined thrust and body moments about a `site` placed at the inertial frame. The `ctrlrange` limits are currently arbitrary and need to be further tuned.
+6. Added `scene.xml` which includes the quadrotor, with a textured ground plane and skybox.
+
+## License
+
+This model is released under the MIT License.
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/README.md:22]: Link the license word to the license, i.e., [MIT License](LICENSE).

Also change "the" to "an".

File: bitcraze_crazyflie_2/cf2.xml [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+      <geom group="3" type="mesh"/>
+    </default>
+  </default>
+
+  <asset>
+    <material name="polished_plastic1_1" specular="0.5" shininess="0.225" rgba="0.631373 0.658824 0.678431 1.000000"/>
+    <material name="polished_gold_1" specular="0.5" shininess="0.225" rgba="0.968627 0.878431 0.600000 1.000000"/>
+    <material name="medium_gloss_plastic_1" specular="0.5" shininess="0.225" rgba="0.109804 0.184314 0.000000 1.000000"/>
+    <material name="Material15_001" specular="0.5" shininess="0.225" rgba="0.792 0.820 0.933 1.000"/>
+    <material name="polypropylene2_1" specular="0.5" shininess="0.225" rgba="1.000000 1.000000 1.000000 1.000000"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:20]: Nit: change all the 1.0000 to 1.

File: bitcraze_crazyflie_2/cf2.xml [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+    <material name="burnished_chrome_1" specular="0.5" shininess="0.225" rgba="0.898039 0.898039 0.898039 1.000000"/>
+
+    <mesh file="cf2_0.obj"/>
+    <mesh file="cf2_1.obj"/>
+    <mesh file="cf2_2.obj"/>
+    <mesh file="cf2_3.obj"/>
+    <mesh file="cf2_4.obj"/>
+    <mesh file="cf2_5.obj"/>
+    <mesh file="cf2_6.obj"/>
+
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:31]: Nit: No need for a newline between the collision and visual objs.

File: bitcraze_crazyflie_2/cf2/cf2.xml [hunk 18]
```diff
@@ -0,0 +1,101 @@
+<mujoco model="cf2">
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2/cf2.xml:1]: Remove this file.

File: bitcraze_crazyflie_2/cf2/cf2_collision_0.obj [hunk 19]
```diff
@@ -0,0 +1,74 @@
+v 0.034218110 -0.031184591 0.007464440
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2/cf2_collision_0.obj:1]: Make sure all OBJ files end with an empty newline (looks like some of them don't).

**Review by kevinzakka — CHANGES_REQUESTED** [review]
Thanks @varadVaidya, left some last few comments.

File: bitcraze_crazyflie_2/cf2.xml [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+    <default class="visual">
+      <geom group="2" type="mesh" contype="0" conaffinity="0"/>
+    </default>
+    <default class="collision">
+      <geom group="3" type="mesh"/>
+    </default>
+  </default>
+
+  <asset>
+    <material name="polished_plastic1_1" specular="0.5" shininess="0.225" rgba="0.631373 0.658824 0.678431 1.000000"/>
```
Comment (kevinzakka) [inline bitcraze_crazyflie_2/cf2.xml:16]: * Can you remove the suffixes (e.g. `_1`) from the material names?
* Can you change "Material15_001" to the actual color?

**Review by kevinzakka — CHANGES_REQUESTED** [review]

**Comment by kevinzakka** [comment]
Hi @varadVaidya, if you don’t have much time to make the changes, I can start the merge and make them myself but need your confirmation.

**Comment by varadVaidya** [comment]
> Hi @varadVaidya, if you don’t have much time to make the changes, I can start the merge and make them myself but need your confirmation.

Hi. I will do the suggested changes by evening IST (GMT +5:30)

**Comment by varadVaidya** [comment]
Hi. I have done the changes you have suggested. Please, let me know if there are any more things needed to be done. 
> if you don’t have much time to make the changes, I can start the merge and make them myself but need your confirmation.

Please, feel free to make any changes required. Just mention them here, so that I can learn from them too.

**Comment by kevinzakka** [comment]
Thanks @varadVaidya, I've started the review process internally!

**Comment by kevinzakka** [comment]
Thank you for your contribution @varadVaidya !

**Pull request merged**